source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+The following sections use a ``cursor`` variable, which is a cursor that
+retrieved all the documents in a collection with the ``Find()`` function.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+Retrieve All Documents
+----------------------
+
+To populate an array with all of your query results, use the ``All()``
+method:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor all
+   :end-before: end cursor all
+
+.. important:: Memory
+
+   This method crashes your application if your query retrieves a number
+   of documents larger than the size of your array. If this occurs,
+   :ref:`retrieve the documents individually <individual-documents-golang>`.
```

Comment:
I think it would be better to just state what this does.

If the number of size of documents returned by your query exceeds available application memory, your program will crash. If you except a large result set, you should consume the cursor iteratively <...>.